### ISSUE TITLE
Feature Request: Add a location search bar

### ISSUE THREAD

**Issue by emanuelfeld** [nypl-spacetime/surveyor#163]
Each time a new photograph loads, the map zooms out to the default position. Panning and zooming becomes tiring pretty quickly. Adding a geocoding search bar would speed things up if you have an idea of the location.

**Comment by bertspaan** [comment]
@emanuelfeld, thanks for your comment! We should indeed add search. And Surveyor included a search bar in its [earlier iterations](https://github.com/nypl-spacetime/surveyor/commit/ed276a480007efb672bd56dbfd30402c061df6cf). We've experimented with the [Mapbox Geocoding API](https://www.mapbox.com/api-documentation/#geocoding) and [Mapzen Search](https://mapzen.com/products/search/). Both APIs didn't provide very satisfying results: not being able to search for just streets (only with house numbers, try searching _"fifth avenue"_ in [Mapbox](https://www.mapbox.com/mapbox-gl-js/example/mapbox-gl-geocoder/)), [searching _"broadway"_ returns Broadway in San Diego](https://mapzen.com/products/search/?query=broadway&endpoint=autocomplete&lng=-73.98073&lat=40.72606&zoom=12), etc. Google's geocoder is much better, however, [its policies](https://developers.google.com/maps/documentation/geocoding/policies) don't allow for using that API on a different, non-Google, map. 

And last but not least, non of these geocoders allow people to search for historical addresses and street names.

But I agree, a search bar which can quickly bring you to a location would be a huge improvement. What should I do?

**Comment by emanuelfeld** [comment]
Options:

1. Use Mapbox, but sanitize people's input to add "New York, NY". Might improve things enough.
2. Try the NYC Department of City Planning's [geocoding API](https://developer.cityofnewyork.us/api/geoclient-api). Searching a street name alone won't work, but intersections and place names do.
3. Switch to a Google map
4. Build a historical geocoder based on https://github.com/nypl-spacetime/etl-nyc-streets (but you're already using a present-day basemap, so I think that if someone doesn't recognize a building or address they'll have a hard time placing it anyways).

**Comment by bertspaan** [comment]
Those are good suggestions, thanks. However:

1. I tried this, improves things slightly, but also makes things confusing and unpredictable. Also, [this collection](https://digitalcollections.nypl.org/collections/detroit-publishing-company-postcards#/?tab=navigation)!
2. Again, only works for NYC... 
3. Would love to, but then I cannot use [Leaflet.GeotagPhoto](github.com/nypl-spacetime/leaflet.GeotagPhoto), and all bars and bus stops and pizza places will be on the map
4. This is what I should do! I'm working on a search API for NYC Space/Time Directory data, but it's not finished yet...

Maybe I'll just reintroduce Mapbox Search, works good enough most of the time. And if people end up in San Diego, they can always go back to NYC.

**Comment by emanuelfeld** [comment]
Oh, I figured Surveyor was only being used for NYC photos. That's what it sounds like from the about page. Plus all the currently included collections seem to be photos of places NYC. Could always change the geocoding tactic if/when you use it for the wider collection.

**Comment by bertspaan** [comment]
You're right, it's almost all NYC now (except a few New Jersey/Westchester/Long Island photos)... But I wanted to keep Surveyor not too NYC-focused.

I'll look some more into adding Mapbox Search soon!

**Comment by emanuelfeld** [comment]
I think they have a bounding box option for geocoding.
